### ISSUE TITLE
[low] Keyboard Navigation: Many Pages: Custom controls cannot be activated with spacebar

### ISSUE THREAD

**Issue by ogumen** [Expensify/App#5652]
## Action Performed:
Prerequisites:
- User has created personal account and is logged into same.
- User has a few chat contacts on the account.

1. Using Chrome, open https://staging.new.expensify.com/
2. Click on "+" button and select "New Group" option.
3. Tab over to any of the contacts displayed and press the Spacebar to try and select the contact for the new group. Note the results, then press Enter. Contact is selected and moved to top of the panel.
4. Tab over to the 'Create Group' button at the bottom of the panel and press the Spacebar. Note the results, then press Enter.

## Expected Result:
Users expect to interact with controls using standard keyboard inputs for particular controls. If custom controls also have non-standard keyboard interactions, users must be notified. The contacts in the list function and are styled as checkboxes. The standard keyboard activation is the Spacebar only. The 'Create Group' button is styled and functions as a form submit button. Pressing either Enter or Spacebar should activate the button.

## Actual Result:
The custom checkboxes are activated with the Enter key only, instead of the Spacebar only. The 'Create Group' button can be activated with Enter but not the Spacebar.

HTML:
`<div tabindex="0" class="css-1dbjc4n r-1awozwy r-1loqt21 r-1otgn73 r-1i6wzkk r-lrvibr" style="background-color: rgb(255, 255, 255); flex-direction: row; justify-content: space-between; padding-left: 20px; padding-right: 20px; text-decoration-line: none; transition-duration: 0s; -webkit-box-orient: horizontal; -webkit-box-direction: normal; -webkit-box-pack: justify;"><div class="css-1dbjc4n r-13awgt0 r-16y2uox r-1777fci" style="height: 64px; min-width: 0px; padding-bottom: 12px; padding-top: 12px; text-decoration-line: none;">...<div tabindex="0" class="css-1dbjc4n r-1loqt21 r-1otgn73 r-13qz1uu"><div class="css-1dbjc4n r-1777fci" style="background-color: rgb(3, 212, 124); border-radius: 8px; border-width: 0px; height: 40px; opacity: 1; padding-right: 12px; padding-left: 12px;"><div dir="auto" class="css-901oao r-lrvibr" style="color: rgb(255, 255, 255); font-family: GTAmericaExp-Regular; font-size: 13px; font-weight: 700; text-align: center;">Create Group</div></div></div>`

Other occurrence:
- Chat: Message action buttons (on hover), Chat session buttons in the history panel, 'Call Me' button in request call overlay, all emoji buttons, Floating action button, Buttons on message bar (emoji, send, add attachment).

## Workaround:
No workaround

## Area issue was found in:
Create new group overlay, Chat page

## Failed WCAG checkpoints
2.1.1

## User impact:
Many impaired individuals who rely on keyboard only navigation may not be able to interact properly with the components when standard keystrokes are not used.

## Suggested resolution:
Using standard HTML controls (i.e. checkbox, <button>) then styling separately with CSS is the easiest and preferred method to ensure controls can be activated using standard keyboard inputs. If that is not possible, script the expected behaviors as per their standard design patterns manually.
https://webaim.org/techniques/keyboard/#testing
https://www.w3.org/TR/wai-aria-practices-1.2/#keyboard-interaction-5
https://www.w3.org/TR/wai-aria-practices-1.2/#keyboard-interaction-3

## Platform:
Where is this issue occurring?

- Web

**Version Number:**
**Reproducible in staging?:**Yes
**Reproducible in production?:**
**Notes/Photos/Videos:** Any additional supporting documentation
**Issue reported by:**
![Bug5185512_Keyboard_Navigation-Chat-Custom_controls_cannot_be_activated_with_spacebar](https://user-images.githubusercontent.com/88733897/136044470-66c5c4ee-609b-4457-aa43-9832e840f4ca.png)

https://user-images.githubusercontent.com/88733897/136044485-686ac5fe-b901-4aa7-8c5c-3debeb69674d.mp4

![Bug5185512_Keyboard_Navigation-Create_New_Group-Custom_controls_cannot_be_activated_with_spacebar-button](https://user-images.githubusercontent.com/88733897/136044499-97f68ca0-8ffd-4111-a653-33900236d11b.png)
![Bug5185512_Keyboard_Navigation-Create_New_Group-Custom_controls_cannot_be_activated_with_spacebar-checkbox](https://user-images.githubusercontent.com/88733897/136044515-581279db-df0c-4baf-a93e-fbf4296cdbde.png)

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)
